# IS 362 - Week 5 Assignment 

In [1]:
# Import pandas

import pandas as pan
from pandas import DataFrame

# Read the information from the airports and weather CSV files into pandas

airports = pan.read_csv ('airports.csv')
weather = pan.read_csv ('weather.csv')

In [2]:
# Display airport data in the dataframe

airports.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

## Question 1

### What is the northernmost airport in the United States?

In [3]:
# Sort airports table by latitude in descending order

airports.sort_values(['lat'], ascending=False)

faa                                        name        lat         lon  \
417   EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230   BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110   AIN                          Wainwright Airport  70.638056 -159.994722   
708   K03                               Wainwright As  70.613378 -159.860350   
152   ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   
...   ...                                         ...        ...         ...   
930   MUE                               Waimea Kohala  20.001328 -155.668108   
1403  WKL                           Waikoloa Heliport  19.913600 -155.864000   
231   BSF                                Bradshaw Aaf  19.760056 -155.553717   
734   KOA                        Kona Intl At Keahole  19.738767 -156.045631   
679   ITO                                   Hilo Intl  19.721375 -155.048469   

       alt  tz dst              tzone  
417    149  -5   A                NaN  
230     44  -9   A  America/Anchorage  
110     41  -9   A  America/Anchorage  
708     35  -9   A  America/Anchorage  
152     96  -9   A  America/Anchorage  
...    ...  ..  ..                ...  
930   2671 -10   A   Pacific/Honolulu  
1403   109 -10   N   Pacific/Honolulu  
231   6190 -10   A   Pacific/Honolulu  
734     47 -10   N   Pacific/Honolulu  
679     38 -10   N   Pacific/Honolulu  

[1458 rows x 8 columns]

The airports table has Dillant-Hopkins Airport as being at the highest latitude; however, it is listed as being in the Eastern time zone and a Google Maps search shows it to be in New Hampshire. 

With the northernmost state of the United States being Alaska, it was obvious that Dillant-Hopkins Airport could not be the northernmost airport. 

Based on this information, the airports table was filtered to only display airports in GMT -9, the time zone for Alaska.

In [4]:
# Display only airports in GMT -9, the time zone for Alaska, which is the northernmost state of the United States

north = airports[(airports.tz == -9.0)]
north

faa                       name        lat         lon   alt  tz dst  \
34    369         Atmautluak Airport  60.866667 -162.273056    18  -9   A   
65    6K8       Tok Junction Airport  63.329444 -142.953611  1639  -9   A   
86    ABL             Ambler Airport  67.106389 -157.857500   334  -9   A   
95    ADK               Adak Airport  51.878000 -176.646000    18  -9   A   
97    ADQ                     Kodiak  57.749967 -152.493856    78  -9   A   
...   ...                        ...        ...         ...   ...  ..  ..   
1416  WTL        Tuntutuliak Airport  60.335278 -162.666944    16  -9   A   
1418  WWP  North Whale Seaplane Base  56.116389 -133.121667     0  -9   A   
1419  WWT             Newtok Airport  60.939167 -164.641111    25  -9   U   
1434  YAK                    Yakutat  59.301200 -139.393700    33  -9   A   
1440  Z84                      Clear  64.301203 -149.120144   552  -9   A   

                  tzone  
34    America/Anchorage  
65    America/Anchorage  
86    America/Anchorage  
95    America/Anchorage  
97    America/Anchorage  
...                 ...  
1416  America/Anchorage  
1418  America/Anchorage  
1419  America/Anchorage  
1434                NaN  
1440  America/Anchorage  

[240 rows x 8 columns]

The airports table was then sorted by latitude in descending order to determine the northernmost airport in the United States.

In [5]:
# Sort airports table by latitude in descending order and return the northernmost airport in the United States

northernmost = north.sort_values(by='lat', ascending = False)
northernmost.head(1)

faa                        name        lat         lon  alt  tz dst  \
230  BRW  Wiley Post Will Rogers Mem  71.285446 -156.766003   44  -9   A   

                 tzone  
230  America/Anchorage

## Question 2

### What is the easternmost airport in the United States?

In [6]:
# Sort airports table by longitude in descending order

airports.sort_values(['lon'], ascending=[False])

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   
...   ...                            ...        ...         ...   ...  ..  ..   
1251  SNP                 St Paul Island  57.167333 -170.220444    63  -9   A   
1281  SVA               Savoonga Airport  63.686400 -170.493000    53  -9   A   
522   GAM                Gambell Airport  63.766800 -171.733000    27  -9   A   
112   AKB                   Atka Airport  52.220278 -174.206389    56  -9   A   
95    ADK                   Adak Airport  51.878000 -176.646000    18  -9   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York  
...                 ...  
1251  America/Anchorage  
1281  America/Anchorage  
522   America/Anchorage  
112   America/Anchorage  
95    America/Anchorage  

[1458 rows x 8 columns]

Since the question is asking for the easternmost airport in the United States, it was safe to assume that the airport would be located in GMT -5, the Eastern time zone.

The airports table shows Dillant-Hopkins Airport as being the easternmost airport in GMT -5; however, it is located in New Hampshire, which is not the easternmost state in the United States. It does not display a valid value in the tzone column.

Based on this information, the airports table was filtered to only display airports in GMT -5, the easternmost time zone in the United States, and to remove aiports with an invalid time zone.

In [7]:
# Display only airports in GMT -5, which is the easternmost time zone in the United States

east = airports[(airports.tz == -5.0) & (airports.tzone).notnull()]
east

faa                            name        lat        lon   alt  tz dst  \
0     04G               Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
3     06N                 Randall Airport  41.431912 -74.391561   523  -5   A   
4     09J           Jekyll Island Airport  31.074472 -81.427778    11  -5   A   
5     0A9  Elizabethton Municipal Airport  36.371222 -82.173417  1593  -5   A   
6     0G6         Williams County Airport  41.467306 -84.506778   730  -5   A   
...   ...                             ...        ...        ...   ...  ..  ..   
1452  ZTY         Boston Back Bay Station  42.347800 -71.075000    20  -5   A   
1454  ZVE          New Haven Rail Station  41.298669 -72.925992     7  -5   A   
1455  ZWI       Wilmington Amtrak Station  39.736667 -75.551667     0  -5   A   
1456  ZWU        Washington Union Station  38.897460 -77.006430    76  -5   A   
1457  ZYP                    Penn Station  40.750500 -73.993500    35  -5   A   

                 tzone  
0     America/New_York  
3     America/New_York  
4     America/New_York  
5     America/New_York  
6     America/New_York  
...                ...  
1452  America/New_York  
1454  America/New_York  
1455  America/New_York  
1456  America/New_York  
1457  America/New_York  

[519 rows x 8 columns]

The airports table was then sorted by longitude in descending order to determine the easternmost airport in the United States.

In [8]:
# Sort airports table by longitude in descending order and return the easternmost airport in the United States

easternmost = east.sort_values(by='lon', ascending = False)
easternmost.head(1)

faa                        name        lat        lon  alt  tz dst  \
444  EPM  Eastport Municipal Airport  44.910111 -67.012694   45  -5   A   

                tzone  
444  America/New_York

## Question 3

### On February 12th, 2013, which New York area airport had the windiest weather?

In [9]:
# Display weather data in the dataframe

weather.head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  wind_speed  \
0    EWR  2013      1    1     1  39.02  26.06  59.37     270.0    10.35702   
1    EWR  2013      1    1     2  39.02  26.96  61.63     250.0     8.05546   
2    EWR  2013      1    1     3  39.02  28.04  64.43     240.0    11.50780   
3    EWR  2013      1    1     4  39.92  28.04  62.21     250.0    12.65858   
4    EWR  2013      1    1     5  39.02  28.04  64.43     260.0    12.65858   

   wind_gust  precip  pressure  visib             time_hour  
0        NaN     0.0    1012.0   10.0  2013-01-01T06:00:00Z  
1        NaN     0.0    1012.3   10.0  2013-01-01T07:00:00Z  
2        NaN     0.0    1012.5   10.0  2013-01-01T08:00:00Z  
3        NaN     0.0    1012.2   10.0  2013-01-01T09:00:00Z  
4        NaN     0.0    1011.9   10.0  2013-01-01T10:00:00Z

In [10]:
# Display weather data only for February 12, 2013 and sort by wind speed in descending order

weather[(weather.year == 2013) & (weather.month == 2) & (weather.day == 12)].sort_values('wind_speed', ascending = False)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
...      ...   ...    ...  ...   ...    ...    ...    ...       ...   
1006     EWR  2013      2   12     0  39.92  39.02  96.55     240.0   
1027     EWR  2013      2   12    21  35.96  24.08  61.67     300.0   
18415    LGA  2013      2   12     0  42.08  37.40  86.89       NaN   
9709     JFK  2013      2   12     0  42.08  37.04  82.17     280.0   
1029     EWR  2013      2   12    23  35.96  23.00  58.94     250.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
...           ...        ...     ...       ...    ...                   ...  
1006      6.90468        NaN     0.0    1006.9   10.0  2013-02-12T05:00:00Z  
1027      5.75390        NaN     0.0    1013.0   10.0  2013-02-13T02:00:00Z  
18415     4.60312        NaN     0.0       NaN   10.0  2013-02-12T05:00:00Z  
9709      4.60312        NaN     0.0    1007.1   10.0  2013-02-12T05:00:00Z  
1029      4.60312        NaN     0.0    1013.7   10.0  2013-02-13T04:00:00Z  

[72 rows x 15 columns]

The fastest wind speed ever recorded was 254mph ([ThoughtCo](https://www.thoughtco.com/fast-wind-speed-recorded-3444498)), so the wind speed of 1048.36058 is an outlier and likely incorrect.

As the wind speed was recorded hourly, it made sense to find the average wind speed in the 24-hour period for each NYC airport.

The outlier was removed and the recorded wind speeds were averaged to determine the average wind speed on February 12, 2013, for each of the NYC area airports.

In [11]:
# Remove outlier and averaged the hourly recorded wind speed to determine average wind speed for the day for each NYC airport 

weather[(weather.year == 2013) & (weather.month == 2) & (weather.day == 12) & (weather.wind_speed < 350)]\
    .groupby('origin').wind_speed.mean()


origin
EWR    13.258987
JFK    14.384750
LGA    14.960140
Name: wind_speed, dtype: float64

## Answers

### Question 1

The northernmost airport in the United States is Wiley Post-Will Rogers Memorial Airport

### Question 2

The easternmost airport in the United State is the Eastport Municipal Airport

### Question 3

Laguardia Airport had the windiest weather of all New York area airports on February 12, 2013.